ocaml/squeezed/lib/squeeze.ml

```diff
@@ -807,18 +800,21 @@ let balance_memory ?fistpoints io =
 (** Return true if the host memory is currently unbalanced and needs rebalancing *)
 let is_host_memory_unbalanced ?fistpoints io =
   let debug_string, host = io.make_host () in
-  let domains = List.map (fun d -> (d.domid, d)) host.domains in
   let _, _, _, result =
     Squeezer.one_iteration ?fistpoints false (is_balanced io) (Squeezer.make ())
       host io.target_host_free_mem_kib (io.gettimeofday ())
   in
   let is_new_target a =
-    let existing = (List.assoc a.action_domid domains).target_kib in
-    a.new_target_kib <> existing
+    let existing_target_kib =
+      host.domains
+      |> DomainSet.find_first_opt (fun d -> a.action_domid = d.domid)
+      |> Option.fold ~none:Int64.minus_one ~some:(fun d -> d.target_kib)
+    in
+    a.new_target_kib <> existing_target_kib
```

Comment:
This can be done without converting None to -1:
```suggestion
      |> Option.map (fun d -> d.target_kib)
    in
    (Some a.new_target_kib) <> existing_target_kib
```